pytorch_lightning/core/lightning.py

```diff
@@ -1085,19 +1086,82 @@ def manual_backward(self, loss: Tensor, optimizer: Optimizer, *args, **kwargs) -
 
+        .. tip:: In manual mode we still automatically accumulate grad over batches if Trainer(accumulate_grad_batches=x) is set
+            and you use `model.manual_optimizer_step(optimizer)`
+
         Example::
 
             def training_step(...):
                 (opt_a, opt_b) = self.optimizers()
                 loss = ...
                 # automatically applies scaling, etc...
                 self.manual_backward(loss, opt_a)
+                self.manual_optimizer_step(opt_a)
         """
         # make sure we're using manual opt
         self._verify_is_manual_optimization('manual_backward')
 
         # backward
+        self._running_manual_optim = True
         self.trainer.train_loop.backward(loss, optimizer, -1, *args, **kwargs)
+        self._running_manual_optim = False
+
+    def manual_optimizer_step(self, optimizer: Optimizer, force_optimizer_step:bool = False, zero_grad: bool = True) -> None:
+        """
+        Call this directly from your training_step when doing optimizations manually.
+        By using this we can ensure that all the proper scaling when using 16-bit etc has been done for you
+
+        .. tip:: In manual mode we still automatically accumulate grad over batches if Trainer(accumulate_grad_batches=x) is set.
+
+        Args:
+            optimizer: Optimizer used to perform `.step()` call
+
+            force_optimizer_step: Whether to force an optimizer step. Could be useful when having 2 optimizers
+                and we wanted to do accumulated gradients for an optimizer but not for the other one.
+                One could put its own logic to force_step.
+
+            zero_grad: Whether to zero_grad the gradients associated with this optimizer.
+                By default, it is set to True. Make sure to call `zero_grad` if set to False.
+
+        Return:
+            None
+
+        Example::
+
+            def training_step(...):
+                (opt_a, opt_b) = self.optimizers()
+                loss = ...
+                # automatically applies scaling, etc...
+                self.manual_backward(loss, opt_a)
+                # This will force an opt.step() even if accumulate_grad_batches is set.
+                self.manual_optimizer_step(opt_a, force_optimizer_step=True)
+
+        Example::
+
+            def training_step(...):
+                (opt_a, opt_b) = self.optimizers()
+                loss = ...
+                # automatically applies scaling, etc...
+                self.manual_backward(loss, opt_a)
+
+                self.manual_optimizer_step(opt_a, zero_grad=False)
+                ... do something with the gradients
+
+                # Make sure to call `zero_grad`
+                opt_a.zero_grad()
+        """
+        # make sure we're using manual opt
+        self._verify_is_manual_optimization('manual_optimizer_step')
+
+        if not self.trainer.train_loop.should_accumulate() or force_optimizer_step:
```

Comment:
I think we're missing TPU support right? Also after reading this and seeing the similarities to the optimizer_step function, is there any overlap that we can reduce code duplication here? I know theres a few additional steps in this function so was just curious